src/core/ext/filters/client_channel/xds/xds_channel.cc

```diff
@@ -30,8 +30,11 @@ grpc_channel_args* ModifyXdsChannelArgs(grpc_channel_args* args) {
 
 grpc_channel* CreateXdsChannel(const XdsBootstrap& bootstrap,
                                const grpc_channel_args& args,
-                               grpc_error** /*error*/) {
-  if (!bootstrap.server().channel_creds.empty()) return nullptr;
+                               grpc_error** error) {
+  if (!bootstrap.server().channel_creds.empty()) {
+    *error = GRPC_ERROR_CREATE_FROM_STATIC_STRING("no credential found");
```

Comment:
This error message is incorrect.  What's happening here is that a credential *is* found, but the gRPC build is insecure, so we don't have the ability to use credentials.

The error message should say something like "credential specified but gRPC not built with security".

Reply:
fixed